### DISCUSSION TITLE
content_types linked to functions

### DISCUSSION THREAD

**Discussion by ghost** [eternnoir/pyTelegramBotAPI#1674] [Q&A]
Please answer these questions before submitting your issue. Thanks!

1. What version of pyTelegramBotAPI are you using?
4.6.1

2. What OS are you using?
Ubuntu 21.10

3. What version of python are you using?
3.9.7

Hi! I am asking users to updload a picture, for which I created a command "Foto". When they do /foto, I ask them to updload a content type "photo". However, they can upload piuctured regardless of the command; they just click on upload a picture and thatś it. How can I link the "upoload photo" to the command, so that they can only upload a picture after writing "/foto"? 

```
[@bot.message_handler(commands=['foto'])
def send_welcome(message, chat_id):
    chat_id = message.chat.id
    send = bot.send_message(message.from_user.id, 'Subí tu foto.')
    bot.register_next_step_handler(send, photo())
    return 

@bot.message_handler(content_types=["photo"])
def photo(message):
    fileID = message.photo[-1].file_id
    logger.info("User_id, fileID", message.from_user.id, fileID)
    file_info = bot.get_file(fileID)
    #print('file.file_path =', file_info.file_path)
    downloaded_file = bot.download_file(file_info.file_path)
    with open(f"./images/{fileID}.jpg", 'wb') as new_file:
        new_file.write(downloaded_file)
    send = bot.send_message(message.from_user.id, 'Foto recibida, gracias!')]
```

**Comment by coder2020official** [comment]
Moved your issue to discussions.

**Comment by coder2020official** [comment]
So, you need to take a photo-input.
To do that, you have two ways:
https://github.com/eternnoir/pyTelegramBotAPI/blob/master/examples/step_example.py
second:
https://github.com/eternnoir/pyTelegramBotAPI/blob/master/examples/custom_states.py

I prefer states, because they are flexible. But choosing is up to you
